common/djangoapps/track/middleware.py

```diff
@@ -137,6 +138,7 @@ def enter_request_context(self, request):
             'session': self.get_session_key(request),
             'user_id': self.get_user_primary_key(request),
             'username': self.get_username(request),
+            'ip': get_ip(request),
```

Comment:
I would like to see that conversion of None to `''` here, we try hard to avoid writing "None" to the tracking logs.